tests/testthat/test-print.R

```diff
@@ -17,3 +17,32 @@ test_that("print.html preserves dependencies for HTML()", {
   result_contents <- readLines(url)
   expect_true(any(grepl("http://example.com/test.js", result_contents)))
 })
+
+test_that("CRLF is properly handled", {
+  txt <- paste(c("x", "y", ""), collapse = "\r\n")
+
+  tmp <- tempfile(fileext = ".txt")
+  on.exit(unlink(tmp))
+
+  writeBin(charToRaw(txt), tmp)
+
+  obj <- tagList(
+    includeHTML(tmp),
+    includeCSS(tmp),
+    includeMarkdown(tmp),
+    includeScript(tmp),
+    includeText(tmp),
+    txt,
+    HTML(txt)
+  )
+
+  out <- tempfile(fileext = ".html")
+  on.exit(unlink(out))
+
+  save_html(obj, out)
+
+  chr <- readChar(out, file.size(out))
+  expect_false(grepl("\r\r\n", chr))
+
+  expect_false(grepl("\r\r\n", as.character(obj)))
```

Comment:
I don't know if it matters, but this test does _not_ fail when I use the CRAN version of htmltools.

Reply:
Hmmm, really? On Windows?

Reply:
Right, on Windows.

```
>   save_html(obj, out)
> 
>   chr <- readChar(out, file.size(out))
>   expect_false(grepl("\r\r\n", chr))
Error: grepl("\r\r\n", chr) isn't false.
> 
>   expect_false(grepl("\r\r\n", as.character(obj)))
> 
```


```
- Session info --------------------------------------------------------------------
 setting  value                       
 version  R version 3.6.1 (2019-07-05)
 os       Windows 10 x64              
 system   x86_64, mingw32             
 ui       RStudio                     
 language (EN)                        
 collate  English_United States.1252  
 ctype    English_United States.1252  
 tz       America/Chicago             
 date     2019-09-04                  

- Packages ------------------------------------------------------------------------
 package     * version date       lib source        
 assertthat    0.2.1   2019-03-21 [1] CRAN (R 3.6.1)
 cli           1.1.0   2019-03-19 [1] CRAN (R 3.6.1)
 crayon        1.3.4   2017-09-16 [1] CRAN (R 3.6.1)
 digest        0.6.20  2019-07-04 [1] CRAN (R 3.6.1)
 htmltools   * 0.3.6   2017-04-28 [1] CRAN (R 3.6.1)
 magrittr      1.5     2014-11-22 [1] CRAN (R 3.6.1)
 markdown      1.1     2019-08-07 [1] CRAN (R 3.6.1)
 R6            2.4.0   2019-02-14 [1] CRAN (R 3.6.1)
 Rcpp          1.0.2   2019-07-25 [1] CRAN (R 3.6.1)
 rlang         0.4.0   2019-06-25 [1] CRAN (R 3.6.1)
 rstudioapi    0.10    2019-03-19 [1] CRAN (R 3.6.1)
 sessioninfo   1.1.1   2018-11-05 [1] CRAN (R 3.6.1)
 testthat    * 2.2.1   2019-07-25 [1] CRAN (R 3.6.1)
 withr         2.1.2   2018-03-15 [1] CRAN (R 3.6.1)
 xfun          0.9     2019-08-21 [1] CRAN (R 3.6.1)
```

Reply:
Ohhh, sorry, I misunderstood. Yeah, the `as.character` test doesn't fail on any version AFAIK, just the other one does.